app/src/main/kotlin/com/wire/android/mapper/MessagePreviewContentMapper.kt

```diff
@@ -106,101 +111,117 @@ fun MessagePreview.uiLastMessageContent(): UILastMessageContent {
+
                     AssetType.ASSET ->
                         UILastMessageContent.SenderWithMessage(userUIText, UIText.StringResource(R.string.last_message_asset))
+
                     AssetType.FILE ->
                         UILastMessageContent.SenderWithMessage(userUIText, UIText.StringResource(R.string.last_message_file))
                 }
+
                 is WithUser.ConversationNameChange -> UILastMessageContent.SenderWithMessage(
                     userUIText,
                     UIText.StringResource(R.string.last_message_change_conversation_name)
                 )
+
                 is WithUser.Knock -> UILastMessageContent.SenderWithMessage(
                     userUIText,
                     UIText.StringResource(R.string.last_message_knock)
                 )
+
                 is WithUser.MemberJoined -> UILastMessageContent.SenderWithMessage(
                     userUIText,
                     UIText.StringResource(R.string.last_message_joined_conversation)
                 )
+
                 is WithUser.MemberLeft -> UILastMessageContent.SenderWithMessage(
                     userUIText,
                     UIText.StringResource(R.string.last_message_left_conversation)
                 )
+
                 is WithUser.MembersAdded -> {
                     val membersAddedContent = (content as WithUser.MembersAdded)
                     val isSelfAdded = membersAddedContent.isSelfUserAdded
+                    val otherUsersSize = membersAddedContent.otherUserIdList.size
 
-                    UILastMessageContent.MultipleMessage(
-                        listOf(
-                            userUIText,
-                            UIText.StringResource(R.string.last_message_added),
-                        )
-                            .plus(if (isSelfAdded) listOf(UIText.StringResource(R.string.member_name_you_label_lowercase)) else listOf())
-                            .plusIf(
-                                { isSelfAdded && membersAddedContent.otherUserIdList.isNotEmpty() },
-                                listOf(UIText.StringResource(R.string.label_and))
-                            )
-                            .plusIf(
-                                { membersAddedContent.otherUserIdList.isNotEmpty() },
-                                UIText.PluralResource(
-                                    R.plurals.last_message_people,
-                                    membersAddedContent.otherUserIdList.size,
-                                    membersAddedContent.otherUserIdList.size
-                                )
-                            )
-                            .plus(UIText.StringResource(R.string.last_message_to_conversation))
-                    )
+                    val previewMessageContent = when {
+                        // This case would never be applicable. If self added self, this will be a MemberJoined
+                        isSelfMessage && isSelfAdded -> {
```

Comment:
As the comment reads, this case is not really reachable, because self user can't be the originator/trigger of the adding message and still add himself. But Kubaz still wanted to add it for security.
Also maybe I can rename those fields to something clearer, like `isSelfUserOriginated` and `isSelfUserAdded` or something similar.